### PR TITLE
chore: Update github actions to remove deprecation warnings

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
Updating our github actions to remove deprecation warnings from ci

### CODE REVIEW

**Comment by mohanarpit** [comment]
@hetunandu Can we also improve the `set-output` invocations? This is also generates a big list of warnings in our Github actions.

File: .github/workflows/integration-tests-command.yml [hunk 1]
```diff
@@ -145,24 +145,24 @@ jobs:
       # Setup Java
       - name: Set up JDK 1.11
         if: steps.run_result.outputs.run_result != 'success'
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: "11.0.10"
```
Comment (sharat87) [inline .github/workflows/integration-tests-command.yml:150]:
```suggestion
        with:
          java-version: "11.0.10"
          distribution: adopt
```

File: .github/workflows/integration-tests-command.yml [hunk 2]
```diff
@@ -969,7 +969,7 @@ jobs:
     steps:
       # Update check run called "package"
       - name: Mark package job as complete
-        uses: actions/github-script@v1
+        uses: actions/github-script@v6
```
Comment (sharat87) [inline .github/workflows/integration-tests-command.yml:972]: This also seems to have a breaking change, https://github.com/actions/github-script#breaking-changes.

So I think we need to change `github.pulls` to `github.rest.pulls`, and `github.checks` to `github.rest.checks`.

(Github is not letting me add suggestion for those lines here).

File: .github/workflows/server-build.yml [hunk 3]
```diff
@@ -87,14 +87,14 @@ jobs:
       # Setup Java
       - name: Set up JDK 1.11
         if: steps.run_result.outputs.run_result != 'success'
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: "11.0.10"
```
Comment (sharat87) [inline .github/workflows/server-build.yml:92]:
```suggestion
        with:
          java-version: "11.0.10"
          distribution: adopt
```

File: .github/workflows/test-build-docker-image.yml [hunk 4]
```diff
@@ -227,14 +227,14 @@ jobs:
       # Setup Java
       - name: Set up JDK 1.11
         if: steps.run_result.outputs.run_result != 'success'
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: "11.0.10"
```
Comment (sharat87) [inline .github/workflows/test-build-docker-image.yml:232]:
```suggestion
        with:
          java-version: "11.0.10"
          distribution: adopt
```

File: .github/workflows/test-build-docker-image.yml [hunk 5]
```diff
@@ -717,24 +717,24 @@ jobs:
...
```
Comment (sharat87) [inline .github/workflows/test-build-docker-image.yml:722]:
```suggestion
        with:
          java-version: "11.0.10"
          distribution: adopt
```

File: .github/workflows/test-build-docker-image.yml [hunk 6]
```diff
@@ -1125,21 +1125,21 @@ jobs:
...
```
Comment (sharat87) [inline .github/workflows/test-build-docker-image.yml:1130]:
```suggestion
        with:
          java-version: "11.0.10"
          distribution: adopt
```

File: .github/workflows/appsmithctl.yml [hunk 7]
```diff
@@ -94,9 +94,10 @@ jobs:
       - name: Set result as failed if there are build failures
         if: failure()
         run: |
-          echo "::set-output name=appsmithctl_run_result::failed" > ~/appsmithctl_run_result
+          echo "appsmithctl_run_result=failed" >> $GITHUB_OUTPUT > ~/appsmithctl_run_result
```
Comment (sharat87) [inline .github/workflows/appsmithctl.yml:97]: I don't think multiple redirection works in `bash`. It works in `zsh`, I think, but not in `bash`. We'll have to use `tee`. I see a few other occurrences of this pattern, let's change them as well, to use `tee`.

```suggestion
          echo "appsmithctl_run_result=failed" | tee ~/appsmithctl_run_result >> $GITHUB_OUTPUT
```

Screenshot 2022-12-10 at 10 41 46 AM

**Review by sharat87 — CHANGES_REQUESTED** [review]

File: .github/workflows/appsmithctl.yml (hunk 7 above)
Comment (hetunandu) [reply .github/workflows/appsmithctl.yml:97]: @sharat87  I tested these changes on a forked repo and it worked as expected

Comment (sharat87) [reply .github/workflows/appsmithctl.yml:97]: Oh, that's interesting then! Cool, go ahead then.

**Review by sharat87 — APPROVED** [review]

**Pull request merged**